### PR TITLE
Create ecommerce only create missing update remove and show orders

### PR DESCRIPTION
**Pull request by JoanSerna**

### CODE REVIEW

File: src/core/entity/category-ecommerce.entity.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  @PrimaryGeneratedColumn()
+  id!: number;
+
+  @Column()
+  name!: string;
+
+  @ManyToOne(() => EcommerceEntity, (ecommerce) => ecommerce.categories, { onDelete: 'CASCADE' })
+  ecommerce!: EcommerceEntity;
+
+  @OneToMany(() => ProductEcommerceEntity, (product) => product.category, { onDelete: 'CASCADE', eager: true })
```
Comment (ruedagato) [inline src/core/entity/category-ecommerce.entity.ts:16]:
```suggestion
  @OneToMany(() => ProductEcommerceEntity, (product) => product.category, { onDelete: 'CASCADE' })
```
I found that this option can generate problem in time respond of the services

File: src/core/services/hotel/hotel.service.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ export class HotelService {
   }
 
   getHotel(hotelId: string) {
-    return this.hotelRepository.findOne({ where: { uid: hotelId }, relations: ['specialZones'] });
+    return this.hotelRepository.findOne({ where: { uid: hotelId }, relations: ['specialZones','ecommerce'] });
```
Comment (ruedagato) [inline src/core/services/hotel/hotel.service.ts:52]: create a service to get e-commerce from a hotel

File: src/ecommerce/services/ecommerce.service.ts [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+      });
+    }
+    return await this.ecommerceRepository.save(request.ecommerce);
+  }
+
+  async getEcommerceById(ecommerceId: number): Promise<EcommerceEntity | undefined> {
+    return await this.ecommerceRepository
+      .findOne(
+        { id: ecommerceId },
+        { relations: [ 'hotel', 'categories', 'categories.products' ] },
```
Comment (ruedagato) [inline src/ecommerce/services/ecommerce.service.ts:88]: is it necessary to return the hotel relation?

File: src/ecommerce/services/ecommerce.service.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+                   to_jsonb(e.*) as employer
+            from ecommerce_leaders el
+                     inner join order_ecommerce oe on el."ecommerceId" = oe."ecommerceId"
+                     inner join order_ecommerce_product_ecommerce oepe on oe.id = oepe."orderId"
+                     inner join product_ecommerce pe on oepe."productId" = pe.id
+                     inner join guest g on oe."guestUid" = g.uid
+                     inner join hotel_booking hb on g."hotelBookingId" = hb.id
+                     inner join room r on hb."roomUid" = r.uid
+                     left join employer e on oe."employerUid" = e.uid
+            where el."employerUid" = 'KBq2TB0nyTlmMwQ3RwhYqrI2VNDW'
```
Comment (ruedagato) [inline src/ecommerce/services/ecommerce.service.ts:243]: the id of the employer would be dynamic

File: src/room/room.module.ts [hunk 5]
```diff
@@ -7,9 +7,10 @@ import { EmployerService } from '../core/services/employer/employer.service';
 import { UserService } from '../core/services/user/user.service';
 import { HotelServiceModule } from '../core/services/hotel/hotelService.module';
 import { TranslateService } from '../translate/translate.service';
+import { EcommerceModule } from '../ecommerce/ecommerce.module';
 
 @Module({
-  imports: [TypeOrmModule.forFeature(entities), HotelServiceModule],
+  imports: [TypeOrmModule.forFeature(entities), HotelServiceModule,EcommerceModule],
```
Comment (ruedagato) [inline src/room/room.module.ts:13]: why it is necessary to import on this module?

File: src/hotel/hotel.module.ts [hunk 6]
```diff
@@ -15,13 +15,15 @@ import { RestaurantEntity } from '../core/entity/restaurant.entity';
...
 import { TranslateService } from '../translate/translate.service';
+import { EcommerceModule } from '../ecommerce/ecommerce.module';
 
 @Module({
   imports: [
     TypeOrmModule.forFeature([...entities, RestaurantEntity]),
     HttpModule,
     HotelServiceModule,
     SpecialZoneServiceModule,
+    EcommerceModule
```
Comment (ruedagato) [inline src/hotel/hotel.module.ts:26]: why it is necessary to import on this module?

File: src/room/room.module.ts (hunk 5 above)
Comment (JoanSerna) [reply src/room/room.module.ts:13]: image

File: src/hotel/hotel.module.ts (hunk 6 above)
Comment (JoanSerna) [reply src/hotel/hotel.module.ts:26]: image

**Review by ruedagato — APPROVED** [review]

**Pull request merged**